### PR TITLE
Create MPT Download

### PR DESCRIPTION
**Pull request by gtimti**
This pull request adds ability to create merchant pricing trend download.

### CODE REVIEW

File: lib/gap_intelligence/client/merchant_pricing_trend_downloads.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module GapIntelligence
+  # @see http://api.gapintelligence.com/api/doc/v1/merchant_pricing_trend_downloads.html
+  module MerchantPricingTrendDownloads
+    # Creates new merchant pricing trend download
+    #
+    # @param owner_id [String,Integer] owner id of the download
+    # @param params [Hash] parameters that will be the JSON payload of the http request
+    # @param options [Hash] options the options to make the request with
+    # @yield [request] The Faraday request
+    # @return [Download] the created download
```
Comment (shavkat-samatov) [inline lib/gap_intelligence/client/merchant_pricing_trend_downloads.rb]: shouldn't it be `# @return [MerchantPricingTrendDownload] the created download`

File: lib/gap_intelligence/client/merchant_pricing_trend_downloads.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    # @return [RequestError] error messages
+    # @see http://api.gapintelligence.com/api/doc/v1/merchant_pricing_trend_downloads/create.html
+    def create_merchant_pricing_trend_download(owner_id, params, options = {}, &block)
+      default_option(options, :record_class, MerchantPricingTrendDownload)
+      perform_request(:post, 'merchant_pricing_trend_downloads', options.merge(body: params.merge({ owner_id: owner_id })), &block)
```
Comment (shavkat-samatov) [inline lib/gap_intelligence/client/merchant_pricing_trend_downloads.rb]: probably a question to @GapIntelligence/dev-team, not only to @gtimti - in Download (not MPTDownload) we send all download params under `download` key, here we send MPT params as they are, they are not under one `merchant_pricing_trend_download` key (or whatever the name should be). Do we need that to be consistent?

File: lib/gap_intelligence/models/merchant_pricing_trend_download.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module GapIntelligence
+  class MerchantPricingTrendDownload < Record
+    attributes :start_at, :end_at, class: Date
+    attributes :categories, :countries, class: Array
+    attributes :merchants, :brands, class: Array
+    attributes :on_ad, :on_promo
+    attributes :custom_file_name, :status
```
Comment (shavkat-samatov) [inline lib/gap_intelligence/models/merchant_pricing_trend_download.rb]: we should probably have `specs` attribute as well

File: spec/factories/merchant_pricing_trend_downloads.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  factory :merchant_pricing_trend_download, class: Hash do
+    start_at '2015-11-24'
+    end_at '2016-02-23'
+    categories 'TVs'
+    countries ['United States']
+    merchants ['Walgreens']
+    brands ['HP']
+    on_ad true
+    on_promo false
+    custom_file_name ''
```
Comment (shavkat-samatov) [inline spec/factories/merchant_pricing_trend_downloads.rb]: missing `specs` attribute (though I know it is not on api side yet)

File: spec/gap/models/merchant_pricing_trend_download_spec.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    end
+
+    it 'has merchants as Array' do
+      expect(download.merchants).to be_an_instance_of(Array)
+    end
+
+    it 'has brands as Array' do
+      expect(download.brands).to be_an_instance_of(Array)
+    end
+
```
Comment (shavkat-samatov) [inline spec/gap/models/merchant_pricing_trend_download_spec.rb]: just wanted to mention that place as well so we have `specs`

File: spec/factories/merchant_pricing_trend_downloads.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+FactoryGirl.define do
+  factory :merchant_pricing_trend_download, class: Hash do
+    start_at '2015-11-24'
+    end_at '2016-02-23'
+    categories 'TVs'
```
Comment (shavkat-samatov) [inline spec/factories/merchant_pricing_trend_downloads.rb]: shouldn't that be an array as well?

File: spec/gap/models/merchant_pricing_trend_download_spec.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    it 'has start_at as Date' do
+      expect(download.start_at).to be_an_instance_of(Date)
+    end
+
+    it 'has end_at as Date' do
+      expect(download.end_at).to be_an_instance_of(Date)
+    end
+
+    it 'has categories as Array' do
+      expect(download).to respond_to(:categories)
```
Comment (shavkat-samatov) [inline spec/gap/models/merchant_pricing_trend_download_spec.rb]: shouldn't that be an array as well? we have categories as array on api side

Comment (shavkat-samatov) [inline spec/gap/models/merchant_pricing_trend_download_spec.rb]: ^ we have it as array at least right now 😄

**Comment by shavkat-samatov** [comment]
@gtimti I just reviewed https://github.com/GapIntelligence/gapi/pull/1050 and realized that probably you did not include specs and price filters as they are not done on PB side and are not aware of the name of param that will be sent from PB, at first I thought you just forgot them

**Comment by gtimti** [comment]
@shavkat-samatov , code updated based on your notes, thank you!

**Comment by shavkat-samatov** [comment]
@gtimti looks gr8️⃣ for me! 👍

**Pull request merged**